### PR TITLE
fix: do not have both --enable-ccache and --disable-ccache set and only RELEASE type has --disable-ccache

### PR DESCRIPTION
**Pull request by zdtsw**
In our current build, after add "--disable-ccache" some jobs showing:
`Completed configuring the version string parameter, config args are now:  --with-vendor-name="Eclipse Adoptium" --with-vendor-url=https://adoptium.net/ --with-vendor-bug-url=https://github.com/adoptium/adoptium-support/issues --with-vendor-vm-bug-url=https://github.com/adoptium/adoptium-support/issues --without-version-opt --without-version-pre --with-version-build=7 --with-vendor-version-string=Temurin-17.0.4+7 --with-boot-jdk=/usr/lib/jvm/jdk-16 --with-debug-level=release --with-native-debug-symbols=external --with-jvm-variants=server,client --with-cacerts-file=/home/jenkins/workspace/build-scripts/jobs/jdk17u/jdk17u-linux-arm-temurin/sbin/../security/cacerts --with-jobs=80  --disable-warnings-as-errors --with-source-date=version --with-build-user=adoptium --disable-ccache --target=armv7l-unknown-linux-gnueabihf --host=armv7l-unknown-linux-gnueabihf --disable-warnings-as-errors --enable-ccache --enable-dtrace`  from https://ci.adoptopenjdk.net/job/build-scripts/job/jobs/job/jdk17u/job/jdk17u-linux-arm-temurin/71/consoleFull which has both flags set

1. move setting for reproducible part after source ${PLATFORM_CONFIG_FILEPATH}.sh
2. remove --enable-ccache when adding --disable-ccache
3. only use --disable-ccache for RELEASE type

Fixes: https://github.com/adoptium/temurin-build/issues/2973

### CODE REVIEW

**Review by karianna — APPROVED** [review]
LGTM although one day we're going to have to rewrite this mess of shell script and extract common code in something like Python :-).  Dreams for later though.

**Comment by zdtsw** [comment]
> LGTM although one day we're going to have to rewrite this mess of shell script and extract common code in something like Python :-). Dreams for later though.

i am still waiting for @andrew-m-leonard to confirm if should have `--disable-ccache `only for "RELEASE" or not. 
Maybe I made a wrong implementation even in the previous commit....
But yes, I agree, the logic is built up way to complex. a refactory is needed but also risky. To maintain logic which adoptium not run/test is a lot of work + old platform e.g freebsd
as for Python :'-(  </3 leave it to another topic

**Comment by andrew-m-leonard** [comment]
@zdtsw yes --disable-ccache for RELEASE=true only, as that's when reproducible builds are enabled

**Comment by zdtsw** [comment]
so i changed mind, to make it simple, do not "+" --disable-ccache in different place, instead of "-" --enable-ccache in the last piece 
also fix the requirement which only need --disable-ccache for RELEASE type build.

**Review by andrew-m-leonard — APPROVED** [review]

**Comment by zdtsw** [comment]
need help to get it merged :D

**Review by gdams — APPROVED** [review]

File: build-farm/set-platform-specific-configurations.sh [hunk 1]
```diff
@@ -118,3 +98,20 @@ else
...
 source "${PLATFORM_CONFIG_FILEPATH}"
+
+# jdk-17 and jdk-19+ support reproducible builds
+if [[ "${JAVA_FEATURE_VERSION}" -ge 19 || "${JAVA_FEATURE_VERSION}" -eq 17 ]]
+then
+    # Enable reproducible builds implicitly with --with-source-date
+    if [ "${RELEASE}" == "true" ]
+    then
+        # Use release date and disable CCache( remove --enable-ccache if exist)
+        export CONFIGURE_ARGS_FOR_ANY_PLATFORM="${CONFIGURE_ARGS_FOR_ANY_PLATFORM//--enable-ccache/} --with-source-date=version --disable-ccache"
```
[the remark this answers is not in the document]
Comment (zdtsw) [reply build-farm/set-platform-specific-configurations.sh:109]: you are right. 
`build.sh` is called after `set-platform-sepcific.configuration.sh`
can we move this block into `mac.sh,` `solaris.sh` and `alpine-linux.sh`
`linux.sh `already has it,
`windows.sh, freebsd.sh and aix.sh `does not have ccache installed.

**Pull request merged**